Tests/Get-BMDeployment.Tests.ps1

```diff
@@ -212,9 +211,177 @@ Describe 'Get-BMDeployment' {
+        $deployment2 = GivenDeployment $build -Stage 'Testing'
+        $deployment3 = GivenDeployment $build2
+        WhenGettingBMDeployment -Pipeline $script:pipeline.Pipeline_Name
+        ThenShouldNotThrowErrors
+        ThenTotalDeploymentsReturned 2
+        ThenDeploymentShouldBeReturned $deployment
+        ThenDeploymentShouldBeReturned $deployment2
+        ThenDeploymentShouldNotBeReturned $deployment3
+    }
+
+    It 'should get a deployment by pipeline stage name' {
+        $build = GivenReleaseBuild $script:releaseAll
+        $build2 = GivenReleaseBuild $script:releaseRelease
+        $deployment = GivenDeployment $build
+        $deployment2 = GivenDeployment $build -Stage 'Testing'
+        $deployment3 = GivenDeployment $build2
+        WhenGettingBMDeployment -Stage 'Testing'
+        ThenShouldNotThrowErrors
+        ThenDeploymentShouldNotBeReturned $deployment
+        ThenDeploymentShouldBeReturned $deployment2
+        ThenDeploymentShouldNotBeReturned $deployment3
+        ThenTotalDeploymentsReturned 1
+    }
+
+    It 'should get a deployment by status' {
+        $build = GivenReleaseBuild $script:releaseAll
+        $build2 = GivenReleaseBuild $script:releaseRelease
+        $deployment = GivenDeployment $build
+        $deployment2 = GivenDeployment $build -Stage 'Testing'
+        $deployment3 = GivenDeployment $build2
+        WhenGettingBMDeployment -Status 'failed'
+        ThenShouldNotThrowErrors
+        ThenDeploymentShouldNotBeReturned $deployment
+        ThenDeploymentShouldBeReturned $deployment2
+        ThenDeploymentShouldNotBeReturned $deployment3
+        ThenTotalDeploymentsReturned 1
+    }
+
+    It 'should get by multiple parameters' {
+        $build = GivenReleaseBuild $script:releaseAll
+        $build2 = GivenReleaseBuild $script:releaseRelease
+        $deployment = GivenDeployment $build
+        $deployment2 = GivenDeployment $build -Stage 'Testing'
+        $deployment3 = GivenDeployment $build2
+        WhenGettingBMDeployment -Release $script:releaseAll -Pipeline 'Testing'
+        ThenShouldNotThrowErrors
+        ThenDeploymentShouldNotBeReturned $deployment
+        ThenDeploymentShouldNotBeReturned $deployment2
+        ThenDeploymentShouldNotBeReturned $deployment3
+        ThenTotalDeploymentsReturned 0
+    }
+
+    It 'should find no deployments' {
+        $build = GivenReleaseBuild $script:releaseAll
+        $build2 = GivenReleaseBuild $script:releaseRelease
+        $deployment = GivenDeployment $build
+        $deployment2 = GivenDeployment $build -Stage 'Testing'
+        $deployment3 = GivenDeployment $build2
+        WhenGettingBMDeployment -Release $script:releaseRelease -Stage 'Testing' -Application $script:app
+        ThenShouldThrowError
+        ThenDeploymentShouldNotBeReturned $deployment
+        ThenDeploymentShouldNotBeReturned $deployment2
+        ThenDeploymentShouldNotBeReturned $deployment3
+        ThenTotalDeploymentsReturned 0

```

Comment:
These two tests assert different behavior. They both result in zero deployments returned, but one of throws an error and the other doesn't?